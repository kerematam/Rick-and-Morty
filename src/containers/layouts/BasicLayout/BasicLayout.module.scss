.root {
  margin: 5px 10% 5px 10%;
  min-width: 400px;
}
