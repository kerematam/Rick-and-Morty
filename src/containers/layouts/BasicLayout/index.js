import React from 'react'
import { Switch, Route, Redirect } from 'react-router-dom'
import CharacterListPage from 'pages/CharacterListPage'
import CharacterPage from 'pages/CharacterPage'
import Header from 'containers/layouts/Header'
import styles from './BasicLayout.module.scss'

const BasicLayout = () => {
  return (
    <div className={styles.root}>
      <Header />
      <div>
        <Switch>
          <Redirect exact from="/" to="/characters" />
          <Route
            path="/characters"
            exact
            render={() => <CharacterListPage />}
          />
          <Route path="/characters/:id" render={() => <CharacterPage />} />
        </Switch>
      </div>
    </div>
  )
}

export default BasicLayout
