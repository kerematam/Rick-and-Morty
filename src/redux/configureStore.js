import { createStore, applyMiddleware, compose } from 'redux'
import createSagaMiddleware from 'redux-saga'
import createReducer from './rootReducers'
import { routerMiddleware } from 'connected-react-router'
import { createInjectorsEnhancer, forceReducerReload } from 'redux-injectors'

function configureStore(initialState = {}, history) {
  let composeEnhancers = compose
  const reduxSagaMonitorOptions = {}

  if (process.env.NODE_ENV !== 'production' && typeof window === 'object') {
    if (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__)
      composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({})
  }

  const sagaMiddleware = createSagaMiddleware(reduxSagaMonitorOptions)
  const { run: runSaga } = sagaMiddleware
  const middlewares = [sagaMiddleware, routerMiddleware(history)]

  const enhancers = [
    applyMiddleware(...middlewares),
    createInjectorsEnhancer({
      createReducer,
      runSaga,
    }),
  ]

  const store = createStore(
    createReducer(),
    initialState,
    composeEnhancers(...enhancers)
  )

  if (module.hot) {
    module.hot.accept('./rootReducers', () => {
      forceReducerReload(store)
    })
  }

  return store
}

export default configureStore
