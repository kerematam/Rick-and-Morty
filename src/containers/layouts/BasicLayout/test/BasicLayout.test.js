import React, { useCallback } from 'react'
import renderer from 'react-test-renderer'
import BasicLayout from '../index'
import { Provider } from 'react-redux'
import configureStore from 'redux/configureStore'
import { HelmetProvider } from 'react-helmet-async'
import { MemoryRouter } from 'react-router-dom'

const setup = (path = '/') => {
  const store = configureStore()
  const wrapper = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter initialEntries={[path]}>
          <HelmetProvider>
            <BasicLayout />
          </HelmetProvider>
        </MemoryRouter>
      </Provider>
    )
    .toJSON()

  return {
    wrapper,
  }
}

describe('BasicLayout', () => {
  test('correctly handles "/" route', () => {
    const { wrapper } = setup('/')
    expect(wrapper).toMatchSnapshot()
  })

  test('correctly handles "/characters/1" route', () => {
    const { wrapper } = setup('/characters/1')
    expect(wrapper).toMatchSnapshot()
  })
})
