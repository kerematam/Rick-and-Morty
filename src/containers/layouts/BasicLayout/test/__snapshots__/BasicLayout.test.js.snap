// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicLayout correctly handles "/" route 1`] = `
<div
  className="root"
>
  <div
    className="root"
  >
    <span
      className="menu_item"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text menu_item_button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root menu_item_icon MuiSvgIcon-fontSizeSmall"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
            />
          </svg>
          <p
            className="MuiTypography-root menu_item_text MuiTypography-body1"
          >
            Characters
          </p>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </span>
    <span
      className="menu_item_delimiter"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 5H11l5 7-5 7h4.5l5-7z"
        />
        <path
          d="M8.5 5H4l5 7-5 7h4.5l5-7z"
        />
      </svg>
    </span>
    <span
      className="menu_item"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text menu_iten_button"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root menu_item_icon MuiSvgIcon-fontSizeSmall"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
            />
          </svg>
          <p
            className="MuiTypography-root menu_item_text MuiTypography-body1"
          />
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </span>
  </div>
  <div>
    <div
      className="root"
    >
      <div>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`BasicLayout correctly handles "/characters/1" route 1`] = `
<div
  className="root"
>
  <div
    className="root"
  >
    <span
      className="menu_item"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text menu_item_button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root menu_item_icon MuiSvgIcon-fontSizeSmall"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
            />
          </svg>
          <p
            className="MuiTypography-root menu_item_text MuiTypography-body1"
          >
            Characters
          </p>
        </span>
      </button>
    </span>
    <span
      className="menu_item_delimiter"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 5H11l5 7-5 7h4.5l5-7z"
        />
        <path
          d="M8.5 5H4l5 7-5 7h4.5l5-7z"
        />
      </svg>
    </span>
    <span
      className="menu_item"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text menu_iten_button"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root menu_item_icon MuiSvgIcon-fontSizeSmall"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
            />
          </svg>
          <p
            className="MuiTypography-root menu_item_text MuiTypography-body1"
          />
        </span>
      </button>
    </span>
  </div>
  <div>
    <div>
      <div
        className="page_container"
      >
        <div
          className="detail_card_container"
        >
          <div
            className="MuiPaper-root detail_card MuiPaper-elevation1 MuiPaper-rounded"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text back_button"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                <span
                  className="MuiButton-startIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                    />
                  </svg>
                </span>
                Characters Page
              </span>
            </button>
          </div>
          <div
            className="MuiPaper-root detail_card MuiPaper-elevation1 MuiPaper-rounded"
          >
            <p
              className="MuiTypography-root detail_item MuiTypography-body1"
            >
              <span
                className="detail_item_key"
              >
                Name
              </span>
              <span
                className="detail_item_value"
              />
            </p>
            <hr
              className="MuiDivider-root"
            />
            <p
              className="MuiTypography-root detail_item MuiTypography-body1"
            >
              <span
                className="detail_item_key"
              >
                Status
              </span>
              <span
                className="detail_item_value"
              />
            </p>
            <hr
              className="MuiDivider-root"
            />
            <p
              className="MuiTypography-root detail_item MuiTypography-body1"
            >
              <span
                className="detail_item_key"
              >
                Origin
              </span>
              <span
                className="detail_item_value"
              />
            </p>
            <hr
              className="MuiDivider-root"
            />
          </div>
          <img
            alt="Character"
          />
        </div>
        <div>
          <div
            className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
          >
            <table
              aria-label="simple table"
              className="MuiTable-root table"
            >
              <thead
                className="MuiTableHead-root"
              >
                <tr
                  className="MuiTableRow-root MuiTableRow-head"
                >
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head"
                    scope="col"
                  >
                    Episode 
                  </th>
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head"
                    scope="col"
                  >
                    Name
                  </th>
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head"
                    scope="col"
                  >
                    Air Date
                  </th>
                </tr>
              </thead>
              <tbody
                className="MuiTableBody-root"
              />
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
